src/shared/utils/cn.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { cn } from './cn'
+
+describe('cn utility', () => {
+  describe('classnames functionality', () => {
+    it('should render styles conditionally', () => {
+      const className = cn('text-white', {
+        'bg-ds-primary-base': false,
+        'bg-ds-pink-tertiary': true,
+      })
+
+      expect(className).toEqual('text-white bg-ds-pink-tertiary')
+    })
+
+    it('should handle many arguments', () => {
+      const className = cn(
+        'text-white text-base',
+        {
+          'bg-ds-primary-base': false,
+          'bg-ds-pink-tertiary': true,
+        },
+        'asdf',
+        { asdf: null },
+        null
+      )
+
+      expect(className).toEqual('text-white text-base bg-ds-pink-tertiary asdf')
```

Comment:
Interesting about 'asdf' and 'asdf:null' resolving to 'asdf', it looks like we're doing ORs as the default?

Reply:
Yea that is interesting. There appears to be [a version of classnames](https://github.com/JedWatson/classnames?tab=readme-ov-file#alternate-dedupe-version) that handles this. What do you think is better?

Reply:
Discussed in Slack. Sticking with default.